code/datums/components/udder.dm

```diff
@@ -60,17 +61,20 @@
  */
 /obj/item/udder
 	name = "udder"
+	///typepath of reagent produced by the udder
+	var/reagent_produced_typepath = /datum/reagent/consumable/milk
 	///how much the udder holds
 	var/size = 50
 	///mob that has the udder component
 	var/mob/living/udder_mob
 	///optional proc to callback to when the udder generates milk
 	var/datum/callback/on_generate_callback
 
-/obj/item/udder/Initialize(mapload, udder_mob, on_generate_callback)
+/obj/item/udder/Initialize(mapload, udder_mob, on_generate_callback, reagent_produced_typepath)
 	src.udder_mob = udder_mob
 	src.on_generate_callback = on_generate_callback
 	create_reagents(size, REAGENT_HOLDER_ALIVE)
+	src.reagent_produced_typepath = reagent_produced_typepath ? reagent_produced_typepath : initial(src.reagent_produced_typepath)
```

Comment:
couldn't you do 
`/obj/item/udder/Initialize(mapload, udder_mob, on_generate_callback, reagent_produced_typepath = /datum/reagent/consumable/milk)`?

Reply:
yes but you see I can only type code or think, not both simultaneously

Reply:
wait no because in the instance where an admin sets too many args and early cancels for instance they will pass the arg as null, which would be different than not passing anything.

Reply:
isn't that a problem with the admin passing arguments into the component? They don't have to specify that they want to change the last argument if they don't want to.

Reply:
ur correct, patching for every admin potential frick up is kinda yiekrs so ill jsut do what u want.